src/views/Farms/components/FarmCard/CardActionsContainer.tsx

```diff
@@ -30,8 +30,17 @@ interface FarmCardActionsProps {
 const CardActions: React.FC<FarmCardActionsProps> = ({ farm, account, addLiquidityUrl }) => {
   const TranslateString = useI18n()
   const [requestedApproval, setRequestedApproval] = useState(false)
-  const { pid, lpAddresses } = useFarmFromSymbol(farm.lpSymbol)
-  const { allowance, tokenBalance, stakedBalance, earnings } = useFarmUser(pid)
+  const { pid, lpAddresses } = farm
+  const {
+    allowance: allowanceAsString = 0,
+    tokenBalance: tokenBalanceAsString = 0,
+    stakedBalance: stakedBalanceAsString = 0,
+    earnings: earningsAsString = 0,
```

Comment:
I think these names are a bit confusing. You have `asString` but they are initialized as numbers.

Reply:
ah yeah, I initially called them "xAsNumber", initialized to `0`, then I remembered that they are stored as strings in Redux, renamed to `xAsStrings` but forgot to change default value to `"0"` 😅  Thankfully BigNumber doesn't care what you put it the constructor.
Also - userData in Redux said to have `BigNumber` type but its incorrect, all userData numbers are strings converted later to BigNumber, its just that we don't have much types defined in Redux part of the code so TS never complained. Anyway, I fixed the types in the table improvements PR I made the other day.